### PR TITLE
JP-3624: Fix a source of errors in resampled spec WCS

### LINKED ISSUE #8484: Bad resample_spec results for MIRI LRS fixed-slit nodded exposures
_Issue [JP-3624](https://jira.stsci.edu/browse/JP-3624) was created on JIRA by [Howard Bushouse](https://jira.stsci.edu/secure/ViewProfile.jspa?name=bushouse):_

Help Desk ticket INC0200735 ([https://stsci.service-now.com/nav_to.do?uri=incident.do?sys_id=21db911a834e8ad05d629da6feaad3fa)] reported an issue with the resample_spec results for obs 5 in program 3786, which is a standard MIRI LRS fixed-slit nodded pair of exposures. Obs 11 in the same program is setup identically and the results for that are fine. The problem appears to be related to the way the output WCS grid is setup for the combined output image from resample_spec, such that the two input (cal) images are being combined on top of one another, without the dither/nod being taken into account. A quick look at the WCS information in the exposures for both obs 5 and 11 shows they are very similar, with all the expected ~1.9 arcsec differences in pointing between the 2 exposures in each observation. So the question is why the 2 exposures from obs 11 are being combined properly, with appropriate shifting/alignment due to the dither offset, while obs 5 is not.

### LINKED ISSUE #8485: Bad resample_spec results for MIRI LRS fixed-slit nodded exposures
_Issue [JP-3624](https://jira.stsci.edu/browse/JP-3624) was created on JIRA by [Howard Bushouse](https://jira.stsci.edu/secure/ViewProfile.jspa?name=bushouse):_

Help Desk ticket INC0200735 ([https://stsci.service-now.com/nav_to.do?uri=incident.do?sys_id=21db911a834e8ad05d629da6feaad3fa](https://stsci.service-now.com/nav_to.do?uri=incident.do?sys_id=21db911a834e8ad05d629da6feaad3fa))) reported an issue with the resample_spec results for obs 5 in program 3786, which is a standard MIRI LRS fixed-slit nodded pair of exposures. Obs 11 in the same program is setup identically and the results for that are fine. The problem appears to be related to the way the output WCS grid is setup for the combined output image from resample_spec, such that the two input (cal) images are being combined on top of one another, without the dither/nod being taken into account. A quick look at the WCS information in the exposures for both obs 5 and 11 shows they are very similar, with all the expected ~1.9 arcsec differences in pointing between the 2 exposures in each observation. So the question is why the 2 exposures from obs 11 are being combined properly, with appropriate shifting/alignment due to the dither offset, while obs 5 is not.

### PR DESCRIPTION
**Pull request by mcara**
Resolves [JP-3624](https://jira.stsci.edu/browse/JP-3624)

Closes #8485
Closes #8484

**Checklist for PR authors (skip items if you don't have permissions or they are not applicable)**
- [x] added entry in `CHANGES.rst` within the relevant release section
- [ ] updated or added relevant tests
- [ ] updated relevant documentation
- [x] added relevant milestone
- [x] added relevant label(s)
- [x] ran regression tests, post a link to the Jenkins job below.
      [How to run regression tests on a PR](https://github.com/spacetelescope/jwst/wiki/Running-Regression-Tests-Against-PR-Branches)
- [ ] All comments are resolved
- [ ] Make sure the JIRA ticket is [resolved properly](https://github.com/spacetelescope/jwst/wiki/How-to-resolve-JIRA-issues)

Regression test: https://plwishmaster.stsci.edu:8081/job/RT/job/JWST-Developers-Pull-Requests/1479/

### CODE REVIEW

File: CHANGES.rst [hunk 1]
```diff
@@ -151,6 +151,10 @@ resample_spec
 
 - Populate the wavelength array in resampled `Slit` and `MultiSlit` models. [#8374]
 
+- Fix a bug resulting in large WCS errors in the resampled image's WCS
+  under certain circumstances (depending on the orientation of the slit on the
```
Comment (emolter) [inline CHANGES.rst:155]: Can you be specific here about what the circumstance was? e.g., 
```suggestion
when the slit was closely aligned with the RA direction
```

**Review by emolter — COMMENTED** [review]
Single-line code change looks fine to me.  Are the regression test differences for `test_miri_lrs_slit_spec2` expected? There appear to be changes in a majority of the pixels, and some of them are several percent.  Also, I assume you have already ensured that this change does indeed allow the exposures mentioned in the original ticket to be combined?  If so, it might be helpful to add a note to the ticket to that effect.

**Review by hbushouse — COMMENTED** [review]
Agree with @emolter that I'd very much like to see a visual before/after comparison of an example where it makes only a slight difference, as well as the example observation that was so far off as to trigger the ticket in the first place.

**Review by hbushouse — APPROVED** [review]
Examples of new results that are provided in the JP ticket look really good. I approve.

**Pull request merged**